### PR TITLE
msat send_spontaneous_payment and lnurl_pay

### PR DESCRIPTION
**Pull request by dangeross**
Breaking changes to the interface methods of send_spontaneous_payment and lnurl_pay

### CODE REVIEW

File: libs/sdk-core/src/breez_services.rs [hunk 1]
```diff
@@ -278,36 +277,32 @@ impl BreezServices {
...
 
     /// Second step of LNURL-pay. The first step is `parse()`, which also validates the LNURL destination
     /// and generates the `LnUrlPayRequestData` payload needed here.
     ///
-    /// This call will validate the given `user_amount_sat` and `comment` against the parameters
+    /// This call will validate the given `amount_msat` and `comment` against the parameters
     /// of the LNURL endpoint (`req_data`). If they match the endpoint requirements, the LNURL payment
     /// is made.
     ///
     /// This method will return an [anyhow::Error] when any validation check fails.
```
Comment (erdemyerebasmaz) [inline libs/sdk-core/src/breez_services.rs:303]: The documentation has to reference LnUrlPayRequest imo.

File: libs/sdk-core/src/models.rs [hunk 2]
```diff
@@ -1164,6 +1179,14 @@ pub enum LnUrlCallbackStatus {
     },
 }
 
+/// Represents a LNURL-pay request.
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct LnUrlPayRequest {
+    pub req_data: LnUrlPayRequestData,
+    pub amount_msat: u64,
+    pub comment: Option<String>,
+}
```
Comment (erdemyerebasmaz) [inline libs/sdk-core/src/models.rs:1188]: We need documentation for the fields of LnUrlPayRequest

**Review by erdemyerebasmaz — CHANGES_REQUESTED** [review]
Looks good, think we're missing changes on documentation comments and I have a suggestion regarding `req_data` vars.

File: libs/sdk-core/src/breez_services.rs (hunk 1 above)
Comment (dangeross) [reply libs/sdk-core/src/breez_services.rs]: Done

File: libs/sdk-core/src/models.rs (hunk 2 above)
Comment (dangeross) [reply libs/sdk-core/src/models.rs]: Done

File: libs/sdk-core/src/models.rs [hunk 3]
```diff
@@ -1182,8 +1182,11 @@ pub enum LnUrlCallbackStatus {
 /// Represents a LNURL-pay request.
 #[derive(Clone, Debug, Serialize, Deserialize)]
 pub struct LnUrlPayRequest {
+    /// The [LnUrlPayRequestData] returned by [BreezServices::parse_input]
     pub req_data: LnUrlPayRequestData,
+    /// The amount in millisatoshis for this payment
     pub amount_msat: u64,
+    /// An option comment for this payment
```
Comment (erdemyerebasmaz) [inline libs/sdk-core/src/models.rs:1189]: NIT: option is more of a rust specific jargon afaik, better to use the word optional here.

**Review by erdemyerebasmaz — APPROVED** [review]
LGTM

One nit comment on wording

File: libs/sdk-core/src/breez_services.rs [hunk 4]
```diff
@@ -345,8 +340,8 @@ impl BreezServices {
                 self.persister.insert_lnurl_payment_external_info(
                     &details.payment_hash,
                     maybe_sa_processed.as_ref(),
-                    Some(req_data.metadata_str),
-                    req_data.ln_address,
+                    Some(req.req_data.metadata_str),
```
[the remark this answers is not in the document]
Comment (erdemyerebasmaz) [reply libs/sdk-core/src/breez_services.rs:347]: Agreed. This change will be handled in #497 

CC: @JssDWt

**Review by erdemyerebasmaz — APPROVED** [review]

**Pull request merged**